### PR TITLE
python-launcher 0.16.0 (new formula)

### PR DESCRIPTION
**Pull request by messense**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

https://github.com/brettcannon/python-launcher

### CODE REVIEW

File: Formula/python-launcher.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  url "https://github.com/brettcannon/python-launcher/archive/refs/tags/v0.16.0.tar.gz"
+  sha256 "03d4a2813c1639264c5e3e6a9961025d7004d1a85deaa8f97bce6010803fa1a8"
+  license "MIT"
+  head "https://github.com/brettcannon/python-launcher.git", branch: "main"
+
+  depends_on "rust" => :build
+  depends_on "python@3.9" => :test
+
+  def install
+    system "cargo", "install", *std_cargo_args
```
Comment (carlocab) [inline Formula/python-launcher.rb:13]: Does this install the fish completions? If not, we can do
```ruby
fish_completion.install "completions/py.fish"
```

Comment (messense) [reply Formula/python-launcher.rb:13]: It doesn't. Added.

File: Formula/python-launcher.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    man1.install "docs/man-page/py.1"
+
+    fish_completion.install "completions/py.fish"
```
Comment (carlocab) [inline Formula/python-launcher.rb:17]:
```suggestion
    system "cargo", "install", *std_cargo_args
    man1.install "docs/man-page/py.1"
    fish_completion.install "completions/py.fish"
```
We don't need all this extra white space. Omitting them makes things just as readable, if not more.

**Comment by messense** [comment]
I've also added `py` as an alias of python-launcher which matches the [Windows Python launcher](https://docs.python.org/3/using/windows.html#launcher) name.

**Comment by nandahkrishna** [comment]
I'm not sure if the `py` alias is desirable (despite this project's similarity to the Windows Python Launcher), perhaps because it's very short, and since the upstream project isn't commonly referred to/installed with such an alias anywhere else (please correct me if this is wrong).

**Comment by messense** [comment]
> I'm not sure if the `py` alias is desirable (despite this project's similarity to the Windows Python Launcher), perhaps because it's very short, and since the upstream project isn't commonly referred to/installed with such an alias anywhere else (please correct me if this is wrong).

It's totally fine to add `py` alias later when desirable.

**Review by nandahkrishna — APPROVED** [review]
Thanks for contributing, @messense!

**Comment by nandahkrishna** [comment]
Oops, didn't notice that `0.16.0` is marked as a pre-release on the GitHub Releases page, I've asked upstream for clarification since this version (and previous ones) have been published to `crates.io` (https://crates.io/crates/python-launcher).

**Comment by nandahkrishna** [comment]
@messense Based on https://github.com/brettcannon/python-launcher/issues/105#issuecomment-860917520 it seems like `python-launcher` isn't stable yet/ready to be packaged. Thank you very much for the PR, but it will have to be closed for now.

Please feel free to resubmit this Formula when a stable version of `python-launcher` is released.

**Pull request closed without merging**